python/dgl/contrib/dis_sampling/sampler.proto

```diff
@@ -0,0 +1,25 @@
+syntax = "proto3";
+
+package sampler;
+
+// The sampler service definition.
+service Sampler {
+  // Sends sub-graph from sampler to trainer
+  rpc SendSubGraph (SamplerRequest) returns (SamplerReply) {}
+}
+
+// The sampler request message
+message SamplerRequest {
+  bytes vertices_id = 1;
+  bytes layer = 2;
+  bytes csr_data = 3;
+  bytes csr_indices = 4;
+  bytes csr_indptr = 5;
+  int32 csr_shape_0 = 6;
+  int32 csr_shape_1 = 7;
+}
```

Comment:
Similar to the above comment, if we are using `GraphIndex` instead of explicit CSR data structure, I would imagine things become:
```protobuf
message SamplerRequest {
  bytes vertices_id = 1;
  bytes layer = 2;
  bytes graph_data = 3;
}
```
, where the third field will be used to create a `GraphIndex` object using `ConvertFromBytes`. 

Also, what are the first the second fields for? I found your demo codes use `mx.nd.contrib.dgl_csr_neighbor_sampling`. Does this protocol only works for that?